File: metrixfolio-fe/types/transaction.ts
export type TransactionType = 'DEPOSIT' | 'WITHDRAWAL';

export interface Transaction {
  id: string;
  type: TransactionType;
  amount: number;
  currency: string;
  date: string; // ISO String (YYYY-MM-DD)
  note?: string;
  created_at?: number;
}

File: metrixfolio-api(deprecated)/src/models/transaction.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum TransactionType {
    DEPOSIT,
    WITHDRAWAL,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Transaction {
    pub id: String,
    #[serde(rename = "type")]
    pub type_: TransactionType,
    pub amount: f64,
    pub currency: String,
    pub date: String,
    pub note: Option<String>,
}
